File: Elearn-DataTier/src/main/java/via/dk/elearn/models/Question.java
package via.dk.elearn.models;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "title", length = 50)
    private String title;

    @Column(name = "description", length = 80)
    private String description;

    @Lob
    @Column(name = "body")
    private String body;

    @Column(name = "url")
    private String url;

    //TODO: Does question have course?
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    private Course course;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    public Question(String body, String title, String url) {
        this.body = body;
        this.title = title;
        this.url = url;
    }

    public Question() {

    }
}

File: Elearn-ApplicationTier/Elearn.Shared/Models/Question.cs
namespace Elearn.Shared.Models;

public class Question
{
    public long Id { get; set; }
    public string Title { get; set; }
    public string Body { get; set; }
    
    public string Description { get; set; }
    public string Url { get; set; }
    public DateTime CreationDate { get; set; }

    public bool CorrectAnswer { get; set; }
    
    public Student Author { get; set; }

    public Question(long id, string title, string body, string description, string url, DateTime creationDate, bool correctAnswer, Student author)
    {
        Id = id;
        Title = title;
        Body = body;
        Description = description;
        Url = url;
        CreationDate = creationDate;
        CorrectAnswer = correctAnswer;
        Author = author;
    }

    public Question(string title, string body, string description, string url, DateTime creationDate, bool correctAnswer, Student author)
    {
        Title = title;
        Body = body;
        Description = description;
        Url = url;
        CreationDate = creationDate;
        CorrectAnswer = correctAnswer;
        Author = author;
    }

    public Question()
    {
    }
    
    public override string ToString()
    {
        return $"Id: {Id}, Title: {Title}, Body: {Body}, Url: {Url}, CreationDate: {CreationDate}, CorrectAnswerId: {CorrectAnswer}, Author: {Author}";
    }
}